esp-hal/src/uart.rs

```diff
@@ -746,6 +751,25 @@ where
         self
     }
 
+    /// Configure DTR pin
+    #[instability::unstable]
+    pub fn with_dtr(mut self, dtr: impl PeripheralOutput<'d>) -> Self {
+        let dtr = dtr.into();
+
+        dtr.apply_output_config(&OutputConfig::default());
+        dtr.set_output_enable(true);
+
+        self.dtr_pin = dtr.connect_with_guard(self.uart.info().dtr_signal);
+
+        self
+    }
+
+    /// Enable RS485 mode
+    pub fn with_rs485(self) -> Self {
```

Comment:
- needs to be `unstable` (if you keep it)
- should be called `into_rs485` and there is no way to disable it, either (if you keep it).
- Maybe this function should be turned into a Config struct field? All other configuration is set there.
- I'm still not sure we want to shoehorn RS485 into the main UART driver. Given that this is pretty minor and also unstable, maybe we can, for now.